cose/key_eddsa.go

```diff
@@ -0,0 +1,81 @@
+package cose
+
+import (
+	"bytes"
+	"crypto"
+	"crypto/ed25519"
+	"fmt"
+
+	"github.com/fxamacker/cbor/v2"
+)
+
+// An EdDSAPublicKey is a public key using EdDSA.
+type EdDSAPublicKey struct {
+	key ed25519.PublicKey
+}
+
+// UnmarshalEdDSAPublicKey unmarshals an EdDSA key using the COSE_Key format.
+func UnmarshalEdDSAPublicKey(raw []byte) (key *EdDSAPublicKey, remaining []byte, err error) {
+	var obj publicKeyStructureOKP
+	decoder := cbor.NewDecoder(bytes.NewReader(raw))
+	err = decoder.Decode(&obj)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	// algorithm is optional, so default to EdDSA
+	if obj.Algorithm == 0 {
+		obj.Algorithm = AlgorithmEdDSA
+	}
+
+	if obj.Type != KeyTypeOctet {
+		return nil, nil, fmt.Errorf("%w: invalid key type", ErrInvalidPublicKey)
+	}
+
+	switch obj.Algorithm {
```

Comment:
Mostly style, but curious why switch statement vs `if` blocks for the following checks?

Reply:
it was how the two other key types were done, but they supported more that one algorithm